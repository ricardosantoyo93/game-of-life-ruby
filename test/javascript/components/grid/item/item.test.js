import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import { Item } from '../../../../../app/javascript/components/grid/item/item';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

describe('Item Component', () => {
    const aliveArray = [
        [true]
    ];

    const options = {
        row: 0,
        col: 0
    };

    test('renders', () => {
        const wrapper = shallow(<Item options={options} aliveArray={aliveArray} />);
        expect(wrapper.exists()).toBe(true);
    });

    it('matches snapshot', () => {
        const tree = renderer.create(<Item options={options} aliveArray={aliveArray} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('calls toggleCell on click', () => {
        const toggleCell = jest.fn();
        const wrapper = mount(<Item options={options} aliveArray={aliveArray} toggleCell={toggleCell} />);

        wrapper.find('div').simulate('click');
        expect(toggleCell).toHaveBeenCalled();
    });
});
