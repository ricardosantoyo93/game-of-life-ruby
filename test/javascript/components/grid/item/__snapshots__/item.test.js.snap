// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component matches snapshot 1`] = `
<div
  className="sc-AxjAm ixIUib"
  onClick={[Function]}
/>
`;
